src/charm.py

```diff
@@ -212,7 +222,7 @@ def _on_install(self, event) -> None:
         # Generate key without a passphrase as traefik does not support it
         # https://github.com/traefik/traefik/pull/6518
         private_key = generate_private_key()
-        self._stored.private_key = private_key.decode()
+        self.secret_store.store_private_key(private_key.decode())
```

Comment:
**Praise**: Delegating to a separate class (which could in the future become a WorkloadManager) is a great idea.